### PR TITLE
Add a pattern Argument

### PR DESCRIPTION
**Pull request by code1mountain**
Add a pattern argument to the NewMessage event. The event will only be passed if the string passes the pattern. This is good for simple command-like interfaces

### CODE REVIEW

File: telethon/events/__init__.py [hunk 1]
```diff
@@ -179,13 +180,14 @@ class NewMessage(_EventBuilder):
...
     def __init__(self, incoming=None, outgoing=None,
-                 chats=None, blacklist_chats=False):
+                 chats=None, blacklist_chats=False, pattern=None):
         if incoming and outgoing:
             raise ValueError('Can only set either incoming or outgoing')
 
         super().__init__(chats=chats, blacklist_chats=blacklist_chats)
         self.incoming = incoming
         self.outgoing = outgoing
+        self.pattern = pattern
```
Comment (Lonami) [inline telethon/events/__init__.py]: `self.pattern`

File: telethon/events/__init__.py [hunk 2]
```diff
@@ -229,13 +231,15 @@ def build(self, update):
...
         # Short-circuit if we let pass all events
-        if all(x is None for x in (self.incoming, self.outgoing, self.chats)):
+        if all(x is None for x in (self.incoming, self.outgoing, self.chats, self.pattern)):
             return event
 
         if self.incoming and event.message.out:
             return
         if self.outgoing and not event.message.out:
             return
+        if self.regex and not re.match(self.regex, event.raw_text):
```
Comment (Lonami) [inline telethon/events/__init__.py]: `self.regex` instead `self.pattern`. Are you sure it works?

**Comment by Lonami** [comment]
Also while you confirm the issue I just mentioned, maybe it's better to `re.compile` the pattern if the user gave a string (they could also give an already compiled regex, maybe acount for that).

File: telethon/events/__init__.py (hunk 2 above)
Comment (Lonami) [reply telethon/events/__init__.py]: Also please don't use `event.raw_text`, access `event.message.message` instead (I'd rather avoid accessing properties when building the event).

Comment (code1mountain) [reply telethon/events/__init__.py]: sorry, I created it in my test version and forgot to rename everything. this should be self.pattern as well

**Comment by Lonami** [comment]
Sorry I did a push force and this pull request is kinda broken right now… Can you please delete your fork, fork it again and paste the changes?

**Pull request closed without merging**